concert/experiments/addons.py

```diff
@@ -362,6 +364,44 @@ async def _write_single_image(self, name, image):
         im_writer.write(image)
 
 
+class PCOTimestampCheck(Addon):
+    def __init__(self, experiment):
+        self._timestamp_checks = {}
+        self._experiment = experiment
+        self.timestamp_incorrect = False
+        super().__init__(experiment.acquisitions)
+
+    def _attach(self):
+        """Attach all acquisitions."""
+        for acq in self.acquisitions:
+            self._timestamp_checks[acq] = self._check_timestamp
+            acq.consumers.append(self._timestamp_checks[acq])
+
+    def _detach(self):
+        """Detach all acquisitions."""
+        for acq in self.acquisitions:
+            acq.consumers.remove(self._timestamp_checks)
+
+    async def _check_timestamp(self, producer):
+        self.timestamp_incorrect = False
+        i = 0
+        async for img in producer:
+            if i == 0:
+                if not isinstance(img, ImageWithMetadata) or (
+                        isinstance(img, ImageWithMetadata) and 'frame_number' not in img.metadata):
+                    self._experiment.log.error("No 'frame_number' present in image."
+                                               "camera.timestamp needs to be set to 'both' or"
+                                               "'binary' to use this addon."
+                                               "Works only with pco cameras.")
+                    self.timestamp_incorrect = True
+                    return
+            if img.metadata['frame_number'] != i + 1:
+                self._experiment.log.error(
+                    f"Frame {i + 1} had wrong frame number {img.metadata['frame_number']}.")
+                self.timestamp_incorrect = True
```

Comment:
So `timestamp_incorrect` could actually mean that there is no timestamp or the order of images is wrong, don't we need to distinguish those? Also, shouldn't an error be raised, at least when the order is wrong? Or it should at least be an option like in C compiler when you turn warnings to errors. I am not sure but for some experiments the order might be crucial I guess...

Reply:
I will add the option to raise an Exception at the end of the last acquistion, if the order was not right. I would also raise it if the timestamp is incorrect, since I guess if you attach this addon, you really want to know if the timestamps are there and in the correct order.

Reply:
See https://github.com/ufo-kit/concert/pull/501/commits/77644755a46c72c50e2c32fe4375fce0b0789b41. The tests need to adjusted accordingly.